### PR TITLE
Adds support for promises

### PR DESCRIPTION
**Pull request by twokul**

### CODE REVIEW

**Comment by twokul** [comment]
@pzuraq

File: addon/components/ui-autocomplete.js [hunk 1]
```diff
@@ -19,6 +21,42 @@ export default Component.extend(OptionListAriaMixin, {
     return generateGuid();
   }),
 
+  /*
+    This computed property wraps a passed promise
+    and allows to watch this promise's state.
+
+    @return {PromiseProxyObject} Promise Proxy
+  */
+  proxyContent: computed('asyncItems', 'items', function() {
```
Comment (pzuraq) [inline addon/components/ui-autocomplete.js]: should be `async-items`?

Comment (twokul) [inline addon/components/ui-autocomplete.js]: I think pascalCase is a standard

Comment (pzuraq) [inline addon/components/ui-autocomplete.js]: Was thinking that component API fields should be dasherized to prepare for angle bracket syntax: `<ui-autocomplete async-items={{myPromise}}>`

I'm ok with either, but we should standardize. Either pascalCase for all external APIs, or dasherized.

**Pull request merged**